Library/Homebrew/utils/github/api.rb

```diff
@@ -129,29 +152,31 @@ def self.keychain_username_password
       return unless GITHUB_PERSONAL_ACCESS_TOKEN_REGEX.match?(github_password)
 
       github_password
-    rescue Errno::EPIPE
-      # The above invocation via `Utils.popen` can fail, causing the pipe to be
-      # prematurely closed (before we can write to it) and thus resulting in a
-      # broken pipe error. The root cause is usually a missing or malfunctioning
-      # `git-credential-osxkeychain` helper.
-      nil
     end
 
     def self.credentials
-      @credentials ||= Homebrew::EnvConfig.github_api_token || keychain_username_password
+      @credentials ||= Homebrew::EnvConfig.github_api_token || github_cli_token || keychain_username_password
```

Comment:
Might be safer to put `github_cli_token` last here, to avoid changing behaviour for users who expect `brew` to check the Keychain.

Reply:
Agreed 👍🏻

Reply:
Hmm, I think I would prefer to invert this behavior as a follow-up I had in mind was to make use of **any** `credential-helper` the user configured by using `git credential fill`. I'm happy to swap the order back here, but in the longer run I think it makes more sense to prefer the GitHub CLI over Git credential helpers for a myriad of reasons (including that the GitHub CLI **may be** the credential helper and it avoids triggering GUI prompts/is a 'better' source).

Reply:
People have historically been fairly twitchy (understandably) about us changing the order here so I think it should wait for a major/minor release so we can put it in the release notes.

```suggestion
      # This is not odeprecated but use that magic string so we put
      # github_cli_token before keychain_username_password on next major/minor release
      @credentials ||= Homebrew::EnvConfig.github_api_token || keychain_username_password || github_cli_token
```

should do it.

Reply:
Yup, happy to switch up the order at some point, but it should wait till we can announce it in release notes.